### PR TITLE
[CELEBORN-1767] Fix occasional errors in UT when creating workers

### PR DESCRIPTION
**Pull request by zaynt4606**
### What changes were proposed in this pull request?
Remove assert in setupWorker for UT which might make UT fail directly.
image

### CODE REVIEW

**Comment by FMX** [comment]
This pr seems will make UT fail faster, why can it fix occasional UT failures?

**Comment by zaynt4606** [comment]
> This pr seems will make UT fail faster, why can it fix occasional UT failures?
If `worker.registered.get()` is false, This pr will wait for the worker to register While assertionError will make the UT job aborted.

File: worker/src/test/scala/org/apache/celeborn/service/deploy/MiniClusterFeature.scala [hunk 1]
```diff
@@ -226,12 +226,14 @@ trait MiniClusterFeature extends Logging {
...
             }
+            if (!workers(i).registered.get()) {
+              throw new IllegalStateException(s"worker $i hasn't been registered")
+            }
           }
         }
-        workerInfos.foreach { case (worker, _) => assert(worker.registered.get()) }
         allWorkersStarted = true
       } catch {
         case ex: Throwable =>
```
Comment (zaynt4606) [inline worker/src/test/scala/org/apache/celeborn/service/deploy/MiniClusterFeature.scala:239]: The wait logic is here~

**Review by FMX — APPROVED** [review]
LGTM. Thanks. Merged into main(v0.6.0).

**Pull request closed without merging**